client/app/hosting/cron/hosting-cron.service.js

```diff
@@ -16,109 +24,118 @@ angular.module('services').service('HostingCron', function hostingCron($q, OvhHt
-    },
-  });
+  getCron(serviceName, id) {
+    return this.OvhHttp.get(['/hosting/web/{serviceName}/cron', id].join('/'), {
+      rootPath: 'apiv6',
+      urlParams: {
+        serviceName,
+      },
+    });
+  }
 
-  /*
-     * Create a new cron
-     */
-  this.createCron = (serviceName, model) => OvhHttp.post('/hosting/web/{serviceName}/cron', {
-    rootPath: 'apiv6',
-    urlParams: {
-      serviceName,
-    },
-    data: {
-      command: model.command,
-      description: model.description || undefined,
-      email: model.email || undefined,
-      frequency: model.frequency,
-      language: model.language,
-      status: model.status,
-    },
-  }).then(
-    (data) => {
-      Hosting.resetCrons();
-      return data;
-    },
-    http => $q.reject(http),
-  );
+  createCron(serviceName, model) {
+    return this.OvhHttp.post('/hosting/web/{serviceName}/cron', {
+      rootPath: 'apiv6',
+      urlParams: {
+        serviceName,
+      },
+      data: {
+        command: model.command,
+        description: model.description || undefined,
+        email: model.email || undefined,
+        frequency: model.frequency,
+        language: model.language,
+        status: model.status,
+      },
+    }).then(
+      (data) => {
+        this.Hosting.resetCrons();
+        return data;
+      },
+      http => this.$q.reject(http),
+    );
+  }
 
-  /*
-     * Delete a cron
-     */
-  this.deleteCron = (serviceName, cronId) => OvhHttp.delete(['/hosting/web/{serviceName}/cron', cronId].join('/'), {
-    rootPath: 'apiv6',
-    urlParams: {
-      serviceName,
-    },
-  }).then(
-    (data) => {
-      Hosting.resetCrons();
+  deleteCron(serviceName, cronId) {
+    return this.OvhHttp.delete(['/hosting/web/{serviceName}/cron', cronId].join('/'), {
+      rootPath: 'apiv6',
+      urlParams: {
+        serviceName,
+      },
+    }).then(
+      (data) => {
+        this.Hosting.resetCrons();
+        return data;
+      },
+      http => this.$q.reject(http),
+    );
+  }
+
+  editCron(serviceName, cronId, model) {
+    return this.OvhHttp.put(['/hosting/web/{serviceName}/cron', cronId].join('/'), {
+      rootPath: 'apiv6',
+      urlParams: {
+        serviceName,
+      },
+      data: {
+        command: model.command.startsWith('./') ? model.command.slice(2) : model.command,
+        description: model.description || undefined,
+        email: model.email || undefined,
+        frequency: model.frequency,
+        language: model.language,
+        status: model.status,
+      },
+    }).then((data) => {
+      this.Hosting.resetCrons();
       return data;
-    },
-    http => $q.reject(http),
-  );
+    });
+  }
+
+  getAvailableLanguage(serviceName) {
+    return this.OvhHttp.get('/hosting/web/{serviceName}/cronAvailableLanguage', {
+      rootPath: 'apiv6',
+      urlParams: {
+        serviceName,
+      },
+    });
+  }
+
+
+  formatLanguage(language) {
+    if (language === OTHER || language === OTHER.toUpperCase()) {
```

Comment:
You can simplify the test (this requires to set the test in your test L129) :
```suggestion
    if (language.toUpperCase() === OTHER.toUpperCase()) {
```